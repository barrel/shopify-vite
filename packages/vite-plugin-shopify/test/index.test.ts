import { describe, expect, it } from 'vitest'
import path from 'node:path'
import { build, normalizePath } from 'vite'
import shopify from '../src'
import fs from 'node:fs/promises'

describe('vite-plugin-shopify', () => {
  it('builds out .liquid files for production', async () => {
    await build({
      logLevel: 'silent',
      plugins: [
        shopify({
          themeRoot: path.join(__dirname, '__fixtures__'),
          sourceCodeDir: path.join(__dirname, '__fixtures__', 'frontend'),
          snippetFile: 'vite-tag.liquid'
        })
      ],
      resolve: {
        alias: {
          '@@': normalizePath(path.resolve(path.join(__dirname, '__fixtures__', 'resources', 'js')))
        }
      }
    })

    const tagsHtml = await fs.readFile(path.join(__dirname, '__fixtures__', 'snippets', 'vite-tag.liquid'), { encoding: 'utf8' })

    expect(tagsHtml).toMatchSnapshot()
  })

  it('builds out .liquid files for production with version numbers', async () => {
    await build({
      logLevel: 'silent',
      plugins: [
        shopify({
          themeRoot: path.join(__dirname, '__fixtures__'),
          sourceCodeDir: path.join(__dirname, '__fixtures__', 'frontend'),
          snippetFile: 'vite-tag.liquid',
          versionNumbers: true
        })
      ],
      resolve: {
        alias: {
          '@@': normalizePath(path.resolve(path.join(__dirname, '__fixtures__', 'resources', 'js')))
        }
      }
    })

    const tagsHtml = await fs.readFile(path.join(__dirname, '__fixtures__', 'snippets', 'vite-tag.liquid'), { encoding: 'utf8' })

    expect(tagsHtml).toMatchSnapshot()
  })

  it('builds out .liquid files for production without module preload tags', async () => {
    await build({
      logLevel: 'silent',
      build: {
        modulePreload: false
      },
      plugins: [
        shopify({
          themeRoot: path.join(__dirname, '__fixtures__'),
          sourceCodeDir: path.join(__dirname, '__fixtures__', 'frontend-no-preload'),
          snippetFile: 'vite-tag.liquid'
        })
      ]
    })

    const tagsHtml = await fs.readFile(path.join(__dirname, '__fixtures__', 'snippets', 'vite-tag.liquid'), { encoding: 'utf8' })

    expect(tagsHtml).toMatchSnapshot()
  })
})
