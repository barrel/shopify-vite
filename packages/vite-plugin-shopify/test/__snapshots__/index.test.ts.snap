// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`vite-plugin-shopify > builds out .liquid files for production 1`] = `
"{% comment %}
  IMPORTANT: This snippet is automatically generated by vite-plugin-shopify.
  Do not attempt to modify this file directly, as any changes will be overwritten by the next build.
{% endcomment %}
{% assign path = vite-tag | replace: '@@/', '../../resources/js/' | replace: '~/', '../' | replace: '@/', '../' %}
{% if path == "/test/__fixtures__/frontend/entrypoints/customers.js" or path == "customers.js" %}
  <script src="{{ 'customers-MM9Bv3NP.js' | asset_url | split: '?' | first }}" type="module" crossorigin="anonymous"></script>
  <link rel="modulepreload" href="{{ 'modulepreload-polyfill-B5Qt9EMX.js' | asset_url | split: '?' | first }}" crossorigin="anonymous">
  <link rel="modulepreload" href="{{ 'foo-CpK_S8xR.js' | asset_url | split: '?' | first }}" crossorigin="anonymous">
  {{ 'foo-Dei2f1sT.css' | asset_url | split: '?' | first | stylesheet_tag: preload: preload_stylesheet }}
  {{ 'customers-CTbC3cH9.css' | asset_url | split: '?' | first | stylesheet_tag: preload: preload_stylesheet }}
{% elsif path == "/test/__fixtures__/frontend/entrypoints/theme.css" or path == "theme.css" %}
  {{ 'theme-y6Yj_vm2.css' | asset_url | split: '?' | first | stylesheet_tag: preload: preload_stylesheet }}
{% elsif path == "/test/__fixtures__/frontend/entrypoints/theme.js" or path == "theme.js" %}
  <script src="{{ 'theme-Df-cUSaP.js' | asset_url | split: '?' | first }}" type="module" crossorigin="anonymous"></script>
  <link rel="modulepreload" href="{{ 'modulepreload-polyfill-B5Qt9EMX.js' | asset_url | split: '?' | first }}" crossorigin="anonymous">
{% endif %}
"
`;

exports[`vite-plugin-shopify > builds out .liquid files for production with version numbers 1`] = `
"{% comment %}
  IMPORTANT: This snippet is automatically generated by vite-plugin-shopify.
  Do not attempt to modify this file directly, as any changes will be overwritten by the next build.
{% endcomment %}
{% assign path = vite-tag | replace: '@@/', '../../resources/js/' | replace: '~/', '../' | replace: '@/', '../' %}
{% if path == "/test/__fixtures__/frontend/entrypoints/customers.js" or path == "customers.js" %}
  <script src="{{ 'customers-MM9Bv3NP.js' | asset_url }}" type="module" crossorigin="anonymous"></script>
  <link rel="modulepreload" href="{{ 'modulepreload-polyfill-B5Qt9EMX.js' | asset_url }}" crossorigin="anonymous">
  <link rel="modulepreload" href="{{ 'foo-CpK_S8xR.js' | asset_url }}" crossorigin="anonymous">
  {{ 'foo-Dei2f1sT.css' | asset_url | stylesheet_tag: preload: preload_stylesheet }}
  {{ 'customers-CTbC3cH9.css' | asset_url | stylesheet_tag: preload: preload_stylesheet }}
{% elsif path == "/test/__fixtures__/frontend/entrypoints/theme.css" or path == "theme.css" %}
  {{ 'theme-y6Yj_vm2.css' | asset_url | stylesheet_tag: preload: preload_stylesheet }}
{% elsif path == "/test/__fixtures__/frontend/entrypoints/theme.js" or path == "theme.js" %}
  <script src="{{ 'theme-Df-cUSaP.js' | asset_url }}" type="module" crossorigin="anonymous"></script>
  <link rel="modulepreload" href="{{ 'modulepreload-polyfill-B5Qt9EMX.js' | asset_url }}" crossorigin="anonymous">
{% endif %}
"
`;

exports[`vite-plugin-shopify > builds out .liquid files for production without module preload tags 1`] = `
"{% comment %}
  IMPORTANT: This snippet is automatically generated by vite-plugin-shopify.
  Do not attempt to modify this file directly, as any changes will be overwritten by the next build.
{% endcomment %}
{% assign path = vite-tag | replace: '~/', '../' | replace: '@/', '../' %}
{% if path == "/test/__fixtures__/frontend-no-preload/entrypoints/customers.js" or path == "customers.js" %}
  <script src="{{ 'customers-DbISg_-d.js' | asset_url | split: '?' | first }}" type="module" crossorigin="anonymous"></script>
  {{ 'foo-Dei2f1sT.css' | asset_url | split: '?' | first | stylesheet_tag: preload: preload_stylesheet }}
  {{ 'customers-CTbC3cH9.css' | asset_url | split: '?' | first | stylesheet_tag: preload: preload_stylesheet }}
{% elsif path == "/test/__fixtures__/frontend-no-preload/entrypoints/theme.css" or path == "theme.css" %}
  {{ 'theme-y6Yj_vm2.css' | asset_url | split: '?' | first | stylesheet_tag: preload: preload_stylesheet }}
{% elsif path == "/test/__fixtures__/frontend-no-preload/entrypoints/theme.js" or path == "theme.js" %}
  <script src="{{ 'theme-lYmnRxhD.js' | asset_url | split: '?' | first }}" type="module" crossorigin="anonymous"></script>
{% endif %}
"
`;
